contracts/SolveNQueens.sol

```diff
@@ -22,16 +22,15 @@ contract SolveNQueens {
         testContract.test(owner);
     }
 
-    function main(bool[][] memory unsolved_board) public returns (bool[][] memory solved_board) {
-        (bool a, bool[][] memory solution) = solveNQ(x, N);
+    function main(bool[4][4] memory unsolved_board) public returns (bool[4][4] memory solved_board) {
```

Comment:
Why is it necessary to strictly limit the size to 4?

Reply:
The code currently throws an error if the dimensions are not specified.